tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem_test.cc

```diff
@@ -93,38 +93,53 @@ class GCSFilesystemTest : public ::testing::Test {
 };
 std::string GCSFilesystemTest::tmp_dir_;
 
-::testing::AssertionResult WriteToServer(const std::string& path, size_t length,
-                                         gcs::Client* gcs_client,
+::testing::AssertionResult WriteToServer(const std::string& path, size_t offset,
+                                         size_t length, gcs::Client* gcs_client,
                                          TF_Status* status) {
   std::string bucket, object;
   ParseGCSPath(path, false, &bucket, &object, status);
-  if (TF_GetCode(status) != TF_OK) {
+  if (TF_GetCode(status) != TF_OK)
     return ::testing::AssertionFailure() << TF_Message(status);
-  }
 
   auto writer = gcs_client->WriteObject(bucket, object);
-  writer.write(content, length);
+  writer.write(content + offset, length);
   writer.Close();
-  if (writer.metadata()) {
+  if (writer.metadata())
     return ::testing::AssertionSuccess();
-  } else {
+  else
     return ::testing::AssertionFailure()
            << writer.metadata().status().message();
-  }
 }
 
-::testing::AssertionResult CompareSubString(int64_t offset, size_t n,
+::testing::AssertionResult CompareSubString(int64_t offset, size_t length,
                                             absl::string_view result,
                                             size_t read) {
   // Result isn't a null-terminated string so we have to wrap it inside a
   // `string_view`
-  if (n == read && content_view.substr(offset, n) ==
-                       absl::string_view(result).substr(0, read)) {
+  if (length == read && content_view.substr(offset, length) ==
+                            absl::string_view(result).substr(0, read))
     return ::testing::AssertionSuccess();
-  } else {
+  else
     return ::testing::AssertionFailure()
            << "Result: " << absl::string_view(result).substr(0, read)
-           << " Read:" << read;
+           << " Read: " << read;
+}
+
+::testing::AssertionResult CompareWithServer(const std::string& path,
+                                             size_t offset, size_t length,
+                                             gcs::Client* gcs_client,
+                                             TF_Status* status) {
+  std::string bucket, object;
+  ParseGCSPath(path, false, &bucket, &object, status);
+  if (TF_GetCode(status) != TF_OK)
+    return ::testing::AssertionFailure() << TF_Message(status);
+
+  auto reader = gcs_client->ReadObject(bucket, object);
+  if (!reader)
+    return ::testing::AssertionFailure() << reader.status().message();
+  else {
+    std::string content{std::istreambuf_iterator<char>{reader}, {}};
+    return CompareSubString(offset, length, content, content.length());
```

Comment:
We should always use braces on both branches of the `if/else` expression.

https://google.github.io/styleguide/cppguide.html#Conditionals

I'm fixing this internally, so nothing to do here but to keep in mind for future PRs.

Reply:
Thank you